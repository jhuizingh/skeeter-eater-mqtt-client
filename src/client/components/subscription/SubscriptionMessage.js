import React from 'react';
import PropTypes from 'prop-types';
import {
  Card, Col, Row, Badge
} from 'react-bootstrap';

export default function SubscriptionMessage(props) {
  const { subscription, message } = props;
  console.log(`message=${JSON.stringify(message)}`);
  const { time, qos, content } = message;
  return (
    <Card className="subscription-message">
      <Card.Header>
        <Row>
          <Col md="auto" className="time text-left pl-1">
            {time}
            <Badge>
              {' QoS: '}
              {qos}
            </Badge>
          </Col>
          <Col>{message.topic === subscription.topic ? '' : message.topic}</Col>
        </Row>
      </Card.Header>
      <Card.Body className="text-left">
        <div>
          <Badge variant="primary">Content: </Badge>
          {` ${content}`}
        </div>
      </Card.Body>
    </Card>
  );
}
SubscriptionMessage.propTypes = {
  subscription: PropTypes.shape({
    topic: PropTypes.string.isRequired
  }).isRequired,
  message: PropTypes.shape({
    time: PropTypes.string.isRequired,
    content: PropTypes.string,
    topic: PropTypes.string.isRequired,
    qos: PropTypes.number
  }).isRequired
};
