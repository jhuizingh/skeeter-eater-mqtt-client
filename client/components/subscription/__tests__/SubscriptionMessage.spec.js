import 'jest-dom/extend-expect';
import React from 'react';
import { render } from 'react-testing-library';
import SubscriptionMessage from '../SubscriptionMessage';

const defaultTopic = 'test/topic/1';
const differentTopic = 'another/completely/different/topic';

const setup = (propOverrides, subscriptionOverrides, messageOverrides) => {
  const props = Object.assign({}, propOverrides);
  props.subscription = Object.assign(
    { topic: defaultTopic },
    subscriptionOverrides
  );
  props.message = Object.assign(
    {
      time: new Date(2019, 3, 1, 10, 13, 27).toString(),
      content: 'here is the value',
      topic: defaultTopic,
      qos: 0
    },
    messageOverrides
  );

  const component = render(<SubscriptionMessage {...props} />);

  return {
    props,
    component
  };
};

describe('SubscriptionMessage Component', () => {
  test('render', () => {
    const { component } = setup();
    console.log(component);

    expect(component.container).toMatchSnapshot();
  });
});

describe('Message topic shows if different than Subscription topic', () => {
  test('Does not show when matching', () => {
    const { component } = setup(
      null,
      { topic: defaultTopic },
      { topic: defaultTopic }
    );

    console.log(component);

    const topicNode = component.getByTestId('subscription-topic');
    console.log(topicNode);

    expect(topicNode).not.toHaveTextContent(differentTopic);

    expect(component.container).toMatchSnapshot();
  });

  test('Does show when not matching', () => {
    const { component } = setup(
      null,
      { topic: defaultTopic },
      { topic: differentTopic }
    );

    console.log(component);

    const topicNode = component.getByTestId('subscription-topic');
    console.log(topicNode);

    expect(topicNode).not.toHaveTextContent(differentTopic);

    expect(component.container).toMatchSnapshot();
  });
});
