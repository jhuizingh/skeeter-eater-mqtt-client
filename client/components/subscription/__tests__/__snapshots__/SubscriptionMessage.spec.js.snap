// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message topic shows if different than Subscription topic Does not show when matching 1`] = `
<div>
  <div
    class="subscription-message card"
  >
    <div
      class="card-header"
    >
      <div
        class="row"
      >
        <div
          class="time text-left pl-1 col-md-auto"
        >
          Mon Apr 01 2019 10:13:27 GMT-0700 (Pacific Daylight Time)
          <span
            class="badge"
          >
             QoS: 
            0
          </span>
        </div>
        <div
          class="col"
        >
          <span
            data-testid="subscription-topic"
          />
        </div>
      </div>
    </div>
    <div
      class="text-left card-body"
    >
      <div>
        <span
          class="badge badge-primary"
        >
          Content: 
        </span>
         here is the value
      </div>
    </div>
  </div>
</div>
`;

exports[`Message topic shows if different than Subscription topic Does show when not matching 1`] = `
<div>
  <div
    class="subscription-message card"
  >
    <div
      class="card-header"
    >
      <div
        class="row"
      >
        <div
          class="time text-left pl-1 col-md-auto"
        >
          Mon Apr 01 2019 10:13:27 GMT-0700 (Pacific Daylight Time)
          <span
            class="badge"
          >
             QoS: 
            0
          </span>
        </div>
        <div
          class="col"
        >
          <span
            data-testid="subscription-topic"
          >
            another/completely/different/topic
          </span>
        </div>
      </div>
    </div>
    <div
      class="text-left card-body"
    >
      <div>
        <span
          class="badge badge-primary"
        >
          Content: 
        </span>
         here is the value
      </div>
    </div>
  </div>
</div>
`;

exports[`SubscriptionMessage Component render 1`] = `
<div>
  <div
    class="subscription-message card"
  >
    <div
      class="card-header"
    >
      <div
        class="row"
      >
        <div
          class="time text-left pl-1 col-md-auto"
        >
          Mon Apr 01 2019 10:13:27 GMT-0700 (Pacific Daylight Time)
          <span
            class="badge"
          >
             QoS: 
            0
          </span>
        </div>
        <div
          class="col"
        >
          <span
            data-testid="subscription-topic"
          />
        </div>
      </div>
    </div>
    <div
      class="text-left card-body"
    >
      <div>
        <span
          class="badge badge-primary"
        >
          Content: 
        </span>
         here is the value
      </div>
    </div>
  </div>
</div>
`;
